project/iterative_sorting.py

```diff
@@ -1,43 +1,79 @@
 # Complete the selection_sort() function below in class with your instructor
+# slection sort
+# First look through all the cards and find the smallest.
+# then move the card to the correct positions in the dataset. Swap it with whatever is at the far left.
+#repeat this sequence till things are done
+# Look for the next smallest card.
+# Don't have to look at the smallest.
+# move it to the correct spot
+
+
+
+import random
 def selection_sort( arr ):
-    # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
-        for i in range(0, len(arr) -1):
-            for k in range(0, len(arr) -1):
-                if arr[k] < arr[smallest_index]:
-                    smallest_index = k
-            temp = arr[smallest_index]
-            arr[smallest_index] = arr[cur_index]
-            arr[cur_index] = temp
-
+        for k in range(cur_index, len(arr)):
+            if arr[k] < arr[smallest_index]:
+                smallest_index = k
+        temp = arr[smallest_index]
```

Comment:
remember we talked about a more compact way of swapping? in this case it would look like so:

arr[smallest_index], arr[curr_index] = arr[curr_index], arr[smallest_index]